crates/services/upgradable-executor/src/instance.rs

```diff
@@ -478,13 +478,16 @@ impl Instance<Relayer> {
         self.add_input_data(input)
     }
 
+    //
     pub fn add_validation_input_data(
         self,
-        block: Block,
+        block: &Block,
         options: ExecutionOptions,
     ) -> ExecutorResult<Instance<InputData>> {
+        // TODO: Can we avoid cloning the block?
```

Comment:
Yeah, you can have two types. One for serialization on our side and one for deserialization on the WASM side. On our side you can pass reference